R/geom-sf.R

```diff
@@ -138,33 +130,37 @@ default_aesthetics <- function(type) {
   }
 }
 
-sf_grob <- function(row, lineend = "butt", linejoin = "round", linemitre = 10) {
+sf_grob <- function(x, lineend = "butt", linejoin = "round", linemitre = 10) {
   # Need to extract geometry out of corresponding list column
-  geometry <- row$geometry[[1]]
-
-  if (inherits(geometry, c("POINT", "MULTIPOINT"))) {
-    row <- modify_list(default_aesthetics("point"), row)
-    gp <- gpar(
-      col = alpha(row$colour, row$alpha),
-      fill = alpha(row$fill, row$alpha),
-      # Stroke is added around the outside of the point
-      fontsize = row$size * .pt + row$stroke * .stroke / 2,
-      lwd = row$stroke * .stroke / 2
-    )
-    sf::st_as_grob(geometry, gp = gp, pch = row$shape)
-  } else {
-    row <- modify_list(default_aesthetics("poly"), row)
-    gp <- gpar(
-      col = row$colour,
-      fill = alpha(row$fill, row$alpha),
-      lwd = row$size * .pt,
-      lty = row$linetype,
-      lineend = lineend,
-      linejoin = linejoin,
-      linemitre = linemitre
-    )
-    sf::st_as_grob(geometry, gp = gp)
-  }
+  geometry <- x$geometry
+  type <- sf_types[sf::st_geometry_type(geometry)]
+  is_point <- type %in% 'point'
```

Comment:
Use `"`, not `'`, as the style guide says: https://style.tidyverse.org/syntax.html#quotes

Reply:
Oops - nice catch. My life is a constant struggle between my own preference and Hadleys ;-)